### PR TITLE
workflows: release promotion job updates

### LINKED ISSUE #4566: workflows: add release promotion job
Now we have a staging build & test workflow we need to complete the final stage of the release pipeline with a manual job to promote to release.

This should push to the DockerHub repo all the images currently staged - no rebuild.
This should push the staging bucket contents to a release bucket and the existing server - plus sign with the release GPG key.

### PR DESCRIPTION
**Pull request by patrick-stephens**
Resolves #4566, tested on https://github.com/patrick-stephens/fluent-bit-release-test/runs/4788880583?check_suite_focus=true using disposable server and bucket.

Confirmed to function with environment secrets only as well:
Screenshot from 2022-01-11 16-34-22

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [NA] Example configuration file for the change
- [NA] Debug log output from testing the change

- [NA] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [NA] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

Verified the packages pass tests as well:
```
$ AWS_URL=https://pat-fluent-bit-release.s3.amazonaws.com ./packaging/testing/smoke/packages/run-package-tests.sh 
$ PACKAGE_TEST=debian11 AWS_URL=https://pat-fluent-bit-release.s3.amazonaws.com ./packaging/testing/smoke/packages/run-package-tests.sh 
$ PACKAGE_TEST=centos8 AWS_URL=https://pat-fluent-bit-release.s3.amazonaws.com ./packaging/testing/smoke/packages/run-package-tests.sh 
```
Added a wrapper script that runs all these which also passes too.
`AWS_URL=https://pat-fluent-bit-release.s3.amazonaws.com ./packaging/testing/smoke/packages/local-test-all.sh`

**Documentation**

- [NA] Documentation required for this feature

**Backporting**

- [NA] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/staging-release.yaml [hunk 1]
```diff
@@ -1,121 +1,125 @@
...
-    runs-on: ubuntu-18.04 # no createrepo on more recent runners
+    - name: Sync to release bucket on S3
+      run: |
+        aws s3 sync packaging/releases/ "s3://${{ secrets.AWS_S3_BUCKET_RELEASE }}" --follow-symlinks --no-progress
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      shell: bash
+
+  # We have two options here:
```
Comment (patrick-stephens) [inline .github/workflows/staging-release.yaml:73]: @niedbalski what's your take here? Option 2 is closer to the current process but needs more manual work to complete. Option 1 provides a sync of the full release bucket but as it grows you will need to filter out the RPM/Debs for the version just built to add or do what you need on the manual server.

File: .github/workflows/staging-release.yaml [hunk 2]
```diff
@@ -1,121 +1,125 @@
...
-          STAGING_REGISTRY: ghcr.io
-          STAGING_IMAGE_NAME: ${{ github.repository }}
-
-          # docker.io/fluent/fluent-bit
-          RELEASE_REGISTRY: docker.io
-          RELEASE_IMAGE_NAME: ${{ secrets.DOCKERHUB_ORGANIZATION }}
+    - name: Upload to build server
+      run: |
+        ssh $USERNAME@$HOST mkdir -p /home/$USERNAME/apt
+        rsync -avz packaging/releases/ $USERNAME@$HOST:/home/$USERNAME/apt
```
Comment (patrick-stephens) [inline .github/workflows/staging-release.yaml:121]: We could also sync to a per-version directory, etc.?

Comment (niedbalski) [reply .github/workflows/staging-release.yaml:121]: shouldn't rsync just sync the delta between what is on the builder vs the server?

Comment (patrick-stephens) [reply .github/workflows/staging-release.yaml:121]: Yup.
My main question is the pull from AWS first of the whole release bucket which will keep growing.

**Comment by niedbalski** [comment]
LGTM

**Pull request merged**